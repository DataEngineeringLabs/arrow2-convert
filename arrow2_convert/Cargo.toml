[package]
name = "arrow2_convert"
version = "0.1.0"
authors = [
    "Jorge Leitao <jorgecarleitao@gmail.com>",
    "Chandra Penke <chandrapenke@gmail.com>"
]
edition = "2021"
license = "Apache-2.0 OR MIT"
keywords = ["Arrow", "arrow2"]
repository = "https://github.com/DataEngineeringLabs/arrow2-derive/arrow2_convert"
description = "Convert between nested rust types and Arrow with arrow2"

[dependencies]
arrow2 = { version = "0.9.1", default-features = false }
arrow2_convert_derive = { version = "0.1.0", path = "../arrow2_convert_derive", optional = true }
chrono = { version = "0.4", default_features = false, features = ["std"] }
err-derive = "0.3"
trybuild = "1.0"

[dev-dependencies]
arrow2_convert_derive = { version = "0.1.0", path = "../arrow2_convert_derive" }

[features]
default = [ "derive" ]
derive = [ "arrow2_convert_derive" ]
