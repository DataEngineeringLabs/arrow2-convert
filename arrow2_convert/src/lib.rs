#![doc = include_str!("../README.md")]
#![deny(missing_docs)]
#![forbid(unsafe_code)]

pub mod deserialize;
pub mod field;
pub mod serialize;

// The proc macro is implemented in derive_internal, and re-exported by this
// crate. This is because a single crate can not define both a proc macro and a
// macro_rules macro.
#[cfg(feature = "arrow2_convert_derive")]
#[doc(hidden)]
pub use arrow2_convert_derive::ArrowField;

// Test README with doctests
#[doc = include_str!("../README.md")]
#[cfg(doctest)]
struct ReadmeDoctests;
