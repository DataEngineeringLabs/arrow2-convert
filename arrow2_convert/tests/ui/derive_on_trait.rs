use arrow2_convert::ArrowField;

#[derive(ArrowField)]
trait Test
{}

fn main() {}
