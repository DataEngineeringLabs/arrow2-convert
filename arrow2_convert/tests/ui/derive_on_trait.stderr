error[E0774]: `derive` may only be applied to `struct`s, `enum`s and `union`s
 --> tests/ui/derive_on_trait.rs:3:1
  |
3 |   #[derive(ArrowField)]
  |   ^^^^^^^^^^^^^^^^^^^^^ not applicable here
4 | / trait Test
5 | | {}
  | |__- not a `struct`, `enum` or `union`

warning: unused import: `arrow2_convert::ArrowField`
 --> tests/ui/derive_on_trait.rs:1:5
  |
1 | use arrow2_convert::ArrowField;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
